xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -120,15 +281,32 @@ ConfigUpdate build() {
   }
 
   /**
-   * Starts resource discovery with xDS protocol. This should be the first method to be called in
-   * this class. It should only be called once.
+   * Shutdown this {@link XdsClient} and release resources.
    */
-  abstract void start();
+  abstract void shutdown();
 
   /**
-   * Stops resource discovery. No method in this class should be called after this point.
+   * Registers a watcher to receive {@link ConfigUpdate} for service with the given hostname and
+   * port.
+   *
+   * <p>Unlike watchers for cluster data and endpoint data, at any point of time at most one config
+   * watcher is allowed.
+   *
+   * @param hostName the host name part of the "xds:" URI for the server name that the gRPC client
+   *     targets for. Must NOT contain port.
+   * @param port the port part of the "xds:" URI for the server name that the gRPC client targets
+   *     for. -1 if not specified.
+   * @param watcher the {@link ConfigWatcher} to receive {@link ConfigUpdate}.
    */
-  abstract void shutdown();
+  void watchConfigData(String hostName, int port, ConfigWatcher watcher) {
```

Comment:
Why not just two argument `(String resourceName, ConfigWatcher watcher)`?

Reply:
The most obvious use case of calling this method is in the xDS resolver, which already separated `hostname` and `port` from the xDS URI. I really do not like the idea of passing in `hostname:port` String and use `URI` API to separate them again in `XdsClientImpl`'s implementation.

Reply:
I think @dapengzhang0 meant the `resourceName` used by TD to identify this resource (i.e. service). I don't know if this string matches `hostName:port` but it looks like `hostName` and `port` is more user-friendly.

Reply:
Maybe I missed something. I did not see you need separate them again in `XdsClientImpl`'s implementation. What I saw is the reverse, you only combined the two again https://github.com/grpc/grpc-java/blob/bd19c8a241a3293bd90315e341f5aa7ff0ef7eb6/xds/src/main/java/io/grpc/xds/XdsClientImpl.java#L140 and not use the two separately anymore.

Reply:
There are two uses: `hostname` and `hostname:port`. `hostname:port` is used in LDS request, `hostname` is used to match VirtualHost domains.